### PR TITLE
Rspec tests for ReceiptsUploadMailbox

### PR DESCRIPTION
**Pull request by maxwofford**
~~Intended to make this a draft, and unsure how to unpublish it. This is still a work in progress.~~

These test succeed locally, and will show up on GitHub Actions CI once https://github.com/hackclub/bank/pull/2577 is merged.

### CODE REVIEW

**Review by albertchae — APPROVED** [review]
The test cases look good and comprehensive

Not blocking for this PR, but have we considered using factories instead of fixtures? One thing I don't like about fixtures is that it forces you to look somewhere besides the test to see what the setup is and why it's relevant to the test (although factories can be prone to this issue too). This article touches on this concept https://enterprisecraftsmanship.com/posts/dry-damp-unit-tests/

Unfortunately we'll tradeoff speed with factories.

This blog post is much more forceful about fixtures 😆 https://semaphoreci.com/blog/2014/01/14/rails-testing-antipatterns-fixtures-and-factories.html `Fixtures are Rails’ default way to prepare and reuse test data. Do not use fixtures.`

Apparently fixtures also skip model validations (partly why they're fast), which makes them harder to maintain long term.

**Comment by maxwofford** [comment]
@albertchae That's a great suggestion– I'll read up on that once I'm back from Railsconf! I was definitely feeling the issues with fixtures not being validated while making this (rewrote some fixtures multiple times b/c I thought they were wrong, then removed those changes from my commits once I realized they weren't).

**Comment by maxwofford** [comment]
Just got verbal review & approval from @garyhtou IRL. Will merge once rebase finishes.

**Pull request merged**
